src/RF/e2bp.cpp

```diff
@@ -1,467 +1,466 @@
+        }
+    }
+
+    return ret;
+}
+
+// Ask a device to give its status
+DeviceStatus E2bp::pollForStatus() {
+    // Using 0 for 'begin' packet seems to do the trick
+    uint8_t buf[PAYLOAD_LENGTH];
+    reset();
+    setupRFModule();
+    getPayload(buf, PL_STATUS);
+    sendPayload(buf);
+    return firstPayloadStatus;
+}
+
+// Get device mode from previous received data from the device
+// This way of detecting devices is not working. Shutter devices may return various values.
+DeviceMode E2bp::getDeviceModeFromRecvData() {
+    switch (answerBuf[0]) {
+        case 0:
+            return ON_OFF;
+        case 1:
+            return DIMMER;
+        // YOKIS MVR500ER - 1F - 42 ou 2F - 42
+        case 0x1e:
+        case 0x2f:
+            return SHUTTER;
+        default:
+            return NO_RCPT;
+    }
+}
+
+bool E2bp::press() { return press(false); }
+
+bool E2bp::press(bool dim) {
+    bool ret = true;
+    if (IS_DEBUG_ENABLED) LOG.println("Button pressing");
+    uint8_t buf[PAYLOAD_LENGTH];
+
+    firstPayloadStatus = UNDEFINED;
+    if (dim)
+        getPayload(buf, PL_DIM);
+    else
+        getPayload(buf, PL_BEGIN);
+    ret = sendPayload(buf);
+
+    if (IS_DEBUG_ENABLED) LOG.println("Button pressed");
+    return ret;
+}
+
+bool E2bp::release() {
+    bool ret = true;
+    if (IS_DEBUG_ENABLED) LOG.println("Button releasing");
+    uint8_t buf[PAYLOAD_LENGTH];
+
+    getPayload(buf, PL_END);
+    ret = sendPayload(buf);
+
+    if (IS_DEBUG_ENABLED) LOG.println("Button released");
+    if (ret && device->getMode() == ON_OFF) {
+        device->setStatus(getLastKnownDeviceStatus());
+    } else if (ret && device->getMode() == DIMMER) {
+        device->setStatus(getLastKnownDeviceStatus());
+        device->toggleStatus();
+    }
+    return ret;
+}
+
+// Press and hold a button
+// Send begin payload and dimming payload while
+// release() has not been called
+bool E2bp::pressAndHoldFor(unsigned long duration) {
+    // Only for dimmers and must be > 700ms
+    if (device->getMode() != DIMMER || duration <= 700) return false;
+
+    reset();
+    setupRFModule();
+    press();
+
+    unsigned long timeout = millis() + duration;
+    unsigned long startTime = millis();
+    // Every second until timeout, send a dim command
+    do {
+        // send payload every 1 second
+        if (millis() - startTime >= 1000) {
+            startTime = millis();
+            reset();
+            setupRFModule();
+            press(true);
+        }
+        yield();
+        delayMicroseconds(1000);  // 1 ms delay
+    } while (millis() <= timeout);
+
+    bool ret = release();
+    // no brightness level available but device is ON
+    device->setBrightness(BRIGHTNESS_MAX);
+    return ret;
+}
+
+// Fill a given buffer with the correct payload and return a pointer to it
+uint8_t* E2bp::getPayload(uint8_t* buf, PayloadType type) {
+    buf[0] = 0x00;
+    buf[1] = device->getMode() == SHUTTER ? 0x06 : 0x04;

```

Comment:
Are you sure it will still be working for normal operation ?
I don't have shutters so I can't test that...

Reply:
Well it's working perfectly with my devices. 0x16 seems to be used for driving all the shutters on the bus, 0x06 is a single device (direct connection)

Reply:
Yes, so it will only work for a single type of configuration. I would rather prefer it to work for single devices.

Reply:
Yes I completely get your point. My problem is in terms of concept. At the moment at the end we are adding "modules", and there is no "bus of modules" object, but when we send 0x1? command it drives all, so there can be (at least) 2 concepts:
1- Issue a "drive & drive pairs" payload for a single device
2- Create a "bus of modules" object

Actually 1 might be more appropriate in terms of concept because the single device is acting as a gateway to others. 
But at the end, for home assistant integration, a "bus of modules" appearing as a single shutter is easier to drive.

Reply:
ok, so if I understand correctly `0x16` is used to drive a  bunch of paired devices while `0x06` drives a single one which is paired in a bus ?
What would be the command to drive a single device which is not paired in a bus then ?

Reply:
As a reminder, this part of the code is a contribution, I actually never tested that as I don't own any Yokis shutters.

Reply:
My devices are connected together in a bus.
I figured out that if I send: 
B9 06 00 00 00 00 ?? 00 00 (for example) one is moving
B9 16 00 00 00 00 ?? 00 00 (for example) all are moving
I always address the device I want to move because of the implementation of the commands.

Now from our discussion I'm thinking that maybe if I address another device (gatewatà, but but the address of the device I want to drive in the command, such as:
B9 06 00 00 <device address> ?? 00 00 sent to the gateway
or maybe:
B9 06 00 20 <device address> ?? 00 00 sent to the gateway
The other device may move? Gateway device propagate the command through the bus?

I didn't tested it. I saw this 0x20 on your firmware but I guess we have different setup.

I'm pretty sure it's possible to test it with other devices, using simple commands such as ON, and
B9 04 or B9 14 to address all.